crates/cheatable-starknet/src/rpc.rs

```diff
@@ -474,9 +482,76 @@ impl CheatableSyscallHandler<'_> {
             response_wrapper.write(vm, &mut self.syscall_handler.syscall_ptr)?;
             return Ok(());
         }
+
+        if let SyscallSelector::CallContract = selector {
+            // Increment, since the selector was peeked into before
+            self.syscall_handler.syscall_ptr += 1;
+            return self.execute_syscall(
+                vm,
+                call_contract_syscall,
+                constants::CALL_CONTRACT_GAS_COST,
+            );
+        }
+
         self.syscall_handler.execute_next_syscall(vm, hint)
     }
 
+    fn execute_syscall<Request, Response, ExecuteCallback>(
```

Comment:
Could you explain how using it fixes the issue? Is it about `execute_inner_call` and invoking `execute_call_entry_point`?

Reply:
Syscall handler when recursively calls contracts. In the previous version we replaced syscall handler only for the top level call and recursive calls have been using original syscall handler without overwritten logic.

Reply:
Okay, so I got it right